tests/expecter.go

```diff
@@ -228,7 +229,9 @@ func ExpectBatchWithValidatedSend(expecter expect.Expecter, batch []expect.Batch
 
 			// Remove the \n since it is translated by the console to \r\n.
 			previousSend = strings.TrimSuffix(previousSend, "\n")
-			bExp.R = fmt.Sprintf("%s%s%s", previousSend, "((?s).*)", bExp.R)
+			bExp.R = fmt.Sprintf("%s%s%s|^\\s?%s*%s", previousSend, "((?s).*)", bExp.R, CRLF, bExp.R)
+			//avoid \r in the middle of a command
+			strings.ReplaceAll(bExp.R, " ", `( |\\r)+`)
```

Comment:
Why do you need 2 backslashes here?

Reply:
I think that this is been complied as a regular epxresison to match `\r` that is the literal `\` and the literal `r`. as this is what apears in the raw console log (example: {"component":"tests","level":"info","msg":"[{1 stty cols \r 500 rows 500\r\n[root@testvmih4q6ltg6m8xqhqsklbzz8gndrnp6nl5q82z87v4mgsg66rwb fedora]#  []}])

as you can see the raw log has literally `\r` written threre. if I'll use just one backslash it will match a carriage return which is something like :

```
carriage
return
```
and not the literal `\r`.
I hope I explained it well

Reply:
I believe those logs you quoted actually make carriage returns explicit by replacing them with `\r` `\n`, but I don't think the console actually contain those as literals. Probably worth double-checking.
For example, I wrote the following recently and used the same logs to get the escape codes: https://github.com/kubevirt/kubevirt/blob/93c2d38d99189915662395fcc390678979296661/tests/expecter.go#L153

Reply:
I understand. I think it failed for me with "\\r" which suppose to be the same as `\r`, but I'll double check.

Reply:
["stty cols 500 rows 500\n"](https://github.com/kubevirt/kubevirt/blob/1546fe02637980e0a07e42f9c915b3fb4a3857d4/tests/login.go#L188) was added especially to avoid the line wrap scenario.
Can you please attach a link to the tests that were failing on it? Maybe it contained sent commands longer than 500 chars?
BTW, I saw that only fedora adds " \r" on line wrap. Cirros adds ""\r\r\n". So adding " \n" is not a general solution.

The fix should be not using `ExpectBatchWithValidatedSend` on `configureConsole`. This method is responsible to configure thing so that `ExpectBatchWithValidatedSend` will work. We cannot use it before/during we configure those values.

Reply:
> ["stty cols 500 rows 500\n"](https://github.com/kubevirt/kubevirt/blob/1546fe02637980e0a07e42f9c915b3fb4a3857d4/tests/login.go#L188) was added especially to avoid the line wrap scenario.
> Can you please attach a link to the tests that were failing on it? Maybe it contained sent commands longer than 500 chars?

[here](https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/4010/pull-kubevirt-e2e-k8s-1.16/1314173459147788288#1:build-log.txt%3A5181) line 5181 if the link won't lead you to it.

> BTW, I saw that only fedora adds " \r" on line wrap. Cirros adds ""\r\r\n". So adding " \n" is not a general solution.
> 

maybe I can change it to `\s` which matches both `\r` and `\n`
> The fix should be not using `ExpectBatchWithValidatedSend` on `configureConsole`. This method is responsible to configure thing so that `ExpectBatchWithValidatedSend` will work. We cannot use it before/during we configure those values.

maybe with that modification it will work, and then it worth using it? what do you think?

Reply:
> > ["stty cols 500 rows 500\n"](https://github.com/kubevirt/kubevirt/blob/1546fe02637980e0a07e42f9c915b3fb4a3857d4/tests/login.go#L188) was added especially to avoid the line wrap scenario.
> > Can you please attach a link to the tests that were failing on it? Maybe it contained sent commands longer than 500 chars?
> 
> [here](https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/4010/pull-kubevirt-e2e-k8s-1.16/1314173459147788288#1:build-log.txt%3A5181) line 5181 if the link won't lead you to it.
> 

As I suspected, it happened in the `configureConsole` method with is doing `stty cols 500 rows 500\n` to prevent it.
`ExpectBatchWithValidatedSend` shouldn't be used for `configureConsole`.

> > BTW, I saw that only fedora adds " \r" on line wrap. Cirros adds ""\r\r\n". So adding " \n" is not a general solution.
> 
> maybe I can change it to `\s` which matches both `\r` and `\n`
> 

I believe you shouldn't change it, you should remove it (the whole commit). If you won't use `ExpectBatchWithValidatedSend` on `configureConsole` you should see this issue anymore.

> > The fix should be not using `ExpectBatchWithValidatedSend` on `configureConsole`. This method is responsible to configure thing so that `ExpectBatchWithValidatedSend` will work. We cannot use it before/during we configure those values.
> 
> maybe with that modification it will work, and then it worth using it? what do you think?

I think this commit should be removed from the PR:)

Reply:
OK. removed the second commit